subsystem_controllers/src/drivers_controller/drivers_controller_node.cpp

```diff
@@ -19,146 +19,145 @@
-
-    if (success)
-    {
-      RCLCPP_INFO_STREAM(get_logger(), "Subsystem able to configure");
-      return CallbackReturn::SUCCESS;
-    }
-    else
-    {
-      RCLCPP_INFO_STREAM(get_logger(), "Subsystem unable to configure");
-      return CallbackReturn::FAILURE;
-    }
+DriversControllerNode::DriversControllerNode(const rclcpp::NodeOptions & options)
+: BaseSubsystemController(options)
+{
+  // Don't automatically trigger state transitions from base class on configure
+  // In this class the managed nodes list first needs to be modified then the transition will be
+  // triggered manually
+  trigger_managed_nodes_configure_from_base_class_ = false;
+
+  config_.startup_duration_ = declare_parameter<int>("startup_duration", config_.startup_duration_);
+  config_.driver_timeout_ =
+    declare_parameter<double>("required_driver_timeout", config_.driver_timeout_);
+
+  // carma-config parameters
+  config_.ros1_ssc_driver_name_ =
+    declare_parameter<std::string>("ros1_ssc_driver_name", config_.ros1_ssc_driver_name_);
+  config_.excluded_namespace_nodes_ = declare_parameter<std::vector<std::string>>(
+    "excluded_namespace_nodes", config_.excluded_namespace_nodes_);
+}
+
+carma_ros2_utils::CallbackReturn DriversControllerNode::handle_on_configure(
+  const rclcpp_lifecycle::State & prev_state)
+{
+  auto base_return = BaseSubsystemController::handle_on_configure(prev_state);
 
+  if (base_return != carma_ros2_utils::CallbackReturn::SUCCESS) {
+    RCLCPP_ERROR(get_logger(), "Drivers Controller could not configure");
+    return base_return;
   }
 
-  carma_ros2_utils::CallbackReturn DriversControllerNode::handle_on_activate(const rclcpp_lifecycle::State &prev_state)
-  {
+  config_ = DriversControllerConfig();
 
-    // Reset to automatically trigger state transitions from base class
-    trigger_managed_nodes_configure_from_base_class_ = true;
+  // Load required plugins and default enabled plugins
+  get_parameter<std::string>("ros1_ssc_driver_name", config_.ros1_ssc_driver_name_);
+  get_parameter<int>("startup_duration", config_.startup_duration_);
+  get_parameter<double>("required_driver_timeout", config_.driver_timeout_);
+  get_parameter<std::vector<std::string>>(
+    "excluded_namespace_nodes", config_.excluded_namespace_nodes_);
 
-    // return only either SUCCESS or FAILURE
-    auto base_return = BaseSubsystemController::handle_on_activate(prev_state); // This will activate all base_managed_nodes
+  RCLCPP_INFO_STREAM(get_logger(), "Config: " << config_);
 
-    if (base_return == CallbackReturn::SUCCESS) {
-      RCLCPP_INFO(get_logger(), "Driver Controller activated!");
-    }
-    else if (base_return == CallbackReturn::FAILURE)
-    {
-      RCLCPP_ERROR(get_logger(), "Driver Controller encountered FAILURE when trying to activate the subsystems... please check which driver failed to activate...");
-    }
+  // Handle fact that parameter vectors cannot be empty
+  if (
```

Comment:
It seems like the base subsystem controller does not handle "excluded_namespace_nodes" being empty. Lets add that here as well